Flake: testdockercfgdeletion

```
github.com/openshift/origin/pkg/serviceaccounts/controllers.TestDockercfgDeletion (from github.com_openshift_origin_pkg_serviceaccounts_controllers)

Failing for the past 1 build (Since Failed#612 )
Took 60 ms.
Stacktrace

=== RUN   TestDockercfgDeletion
--- FAIL: TestDockercfgDeletion (0.06s)
	deleted_dockercfg_secrets_test.go:81: deleted dockercfg secret without serviceaccount: Expected {{default get {  serviceaccounts} } default}, got {{ list {  secrets} } { }}
	deleted_dockercfg_secrets_test.go:81: deleted dockercfg secret without serviceaccount: Expected {{default delete {  secrets} } token-secret-1}, got {{ watch {  secrets} } {  }}
	deleted_dockercfg_secrets_test.go:75: deleted dockercfg secret without serviceaccount: 2 unexpected actions: [{ActionImpl:{Namespace:default Verb:get Resource:{Group: Version: Resource:serviceaccounts} Subresource:} Name:default} {ActionImpl:{Namespace:default Verb:delete Resource:{Group: Version: Resource:secrets} Subresource:} Name:token-secret-1}]
```

from https://ci.openshift.redhat.com/jenkins/job/test_branch_origin/612/testReport/github/com_openshift_origin_pkg_serviceaccounts_controllers/TestDockercfgDeletion/

Caused by my change to the controllers

--- a/pkg/serviceaccounts/controllers/deleted_dockercfg_secrets_test.go
+++ b/pkg/serviceaccounts/controllers/deleted_dockercfg_secrets_test.go
@@ -8,6 +8,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	clientgotesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
 	informers "k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion"
@@ -65,6 +66,10 @@ func TestDockercfgDeletion(t *testing.T) {
 		)
 		stopCh := make(chan struct{})
 		informerFactory.Start(stopCh)
+		if !cache.WaitForCacheSync(stopCh, controller.secretController.HasSynced) {
+			t.Fatalf("unable to reach cache sync")
+		}
+		client.ClearActions()
 
 		if tc.DeletedSecret != nil {
 			controller.secretDeleted(tc.DeletedSecret)

